drivers/bluetooth/hci/h4.c

```diff
@@ -37,6 +37,9 @@ LOG_MODULE_REGISTER(bt_driver);
 #define H4_EVT  0x04
 #define H4_ISO  0x05
 
+BUILD_ASSERT(DT_PROP(DT_CHOSEN(zephyr_bt_uart), hw_flow_control) == 1,
+		"hw_flow_control must be enabled for HCI H4 UART");
```

Comment:
I'm not sure this should be a hard assert. There can be a loud warning, sure, but I'm not sure we should prevent someone trying out the H4 driver e.g. if they're missing the physical flow control lines in their setup.

Reply:
Agree with @jhedberg, rather than the BUILD_ASSERT requested by @decsny, let's use a warning. It gives the user the option to enable this feature (and get full reliability), or not and accept the limitations that may come about (depending on their use case).

Reply:
yes, i see many Twister tests start fail. I'll move this 'BUILD_ASSERT...' in to cyw43xxx.c driver, where hw_flow_control for sure must be enabled.

Reply:
> yes, i see many Twister tests start fail. I'll move this 'BUILD_ASSERT...' in to cyw43xxx.c driver, where hw_flow_control for sure must be enabled.

which is what I meant originally